### ISSUE TITLE
Ember 2.8.0 error while hitting `view.parentView = null;` in `cleanupRenderNode` in ember-htmlbars

### ISSUE THREAD

**Issue by josemariaruiz** [emberjs/ember.js#14266]
Hi,

I upgraded my app to 2.8.0 and now I get this error in the browser debugger.

```ember.debug.js:6778 Uncaught Error: Assertion Failed: You must use Ember.set() to set the `parentView` property (of <spa@component:md-card-content::ember940>) to `null`.```

This is interesting, as it's EmberJS telling me that something cannot be done (setting `parentView`  directly) and yet `ember-htmlbars` does it.

Check 

```
enifed('ember-htmlbars/hooks/cleanup-render-node', ['exports'], function (exports) {
  /**
  @module ember
  @submodule ember-htmlbars
  */

  'use strict';

  exports.default = cleanupRenderNode;

  function cleanupRenderNode(renderNode) {
    var view = renderNode.emberView;
    if (view) {
      view.renderer.willDestroyElement(view);
      view.ownerView._destroyingSubtreeForView.push(function (env) {
        view._transitionTo('destroying'); // unregisters view
        // prevents rerender and scheduling
        view._renderNode = null;
        renderNode.emberView = null;

        view.renderer.didDestroyElement(view);

        if (view.parentView && view.parentView === env.view) {
          view.parentView.removeChild(view);
        }
        view.parentView = null;

        view._transitionTo('preRender');
      });
    }

    if (renderNode.cleanup) {
      renderNode.cleanup();
    }
  }
});
```

**Comment by josemariaruiz** [comment]
I traced the problem to the bower module (I don't know if that's the name) that you install with `bower install`. 

The issue is described in this issue report: `https://github.com/components/ember/issues/77`

**Comment by rwjblue** [comment]
This is a duplicate of https://github.com/emberjs/ember.js/issues/14260. As I mentioned in that issue, you should generally not be observing `parentView`. Can you provide a reproduction / example of what you are doing?

**Comment by josemariaruiz** [comment]
A `git grep` in my source code shows no use of `parentView` so I assume that the problem in in a third party.

I'm not doing anything strange in particular, app relays on `ember-cli-materialize` and it has a component called `md-card-content`. 

Running the same code with EmberJS 2.7.0 (that version works flawlessly) I've got this warning:

```DEPRECATION: The `activator` property of `<spa@component:md-card-reveal::ember875>` is an `Ember.Binding` connected to `parentView.activator`, but `Ember.Binding` is deprecated. Consider using an `alias` computed property instead. [deprecation id: ember-metal.binding] See http://emberjs.com/deprecations/v2.x#toc_ember-binding for more details.```

**Comment by rwjblue** [comment]
Wherever md-card-reveal is defined, is where the observer on parentView is being used.

**Comment by josemariaruiz** [comment]
For instance, this code?
```
  title: alias('parentView.title'),
  titleClass: alias('parentView.titleClass'),
  activator: alias('parentView.activator'),
```

**Comment by rwjblue** [comment]
Yes, exactly.

**Comment by rwjblue** [comment]
Closed by #14281.
